### PR TITLE
Finalize Javadoc work in extension-base

### PR DESCRIPTION
**Pull request by alzimmermsft**
Finalizes fixes for Javadoc in `extension-base` and adds a configuration into the parent POM to allow child libraries to enable failing Javadocs on warnings to ensure `extension-base`'s Javadocs are kept up-to-date.

### CODE REVIEW

File: extension-base/src/main/java/com/azure/autorest/extension/base/plugin/NewPlugin.java [hunk 1]
```diff
@@ -85,10 +88,28 @@ public <T> T getValue(Type type, String key) {
         return connection.request(jsonMapper.constructType(type), "GetValue", sessionId, key);
     }
 
+//    /**
```
Comment (srnagar) [inline extension-base/src/main/java/com/azure/autorest/extension/base/plugin/NewPlugin.java:91]: We should remove this commented out code entirely.

File: preprocessor/src/main/java/com/azure/autorest/preprocessor/Preprocessor.java [hunk 2]
```diff
@@ -225,6 +225,16 @@ public <T> T getValue(Type type, String key) {
...
+//    @Override
+//    public <K, V> Map<K, V> getMapValue(Class<K> keyType, Class<V> valueType, String key) {
+//        return wrappedPlugin.getMapValue(keyType, valueType, key);
+//    }
+//
+//    @Override
+//    public <T> List<T> getListValue(Class<T> valueType, String key) {
+//        return wrappedPlugin.getListValue(valueType, key);
+//    }
+
```
Comment (srnagar) [inline preprocessor/src/main/java/com/azure/autorest/preprocessor/Preprocessor.java:237]: Why do we need this commented code?

**Review by srnagar — APPROVED** [review]

**Review by weidongxu-microsoft — APPROVED** [review]
LGTM

Also unsure why this is added.
https://github.com/Azure/autorest.java/pull/2585/files#r1505201091

File: preprocessor/src/main/java/com/azure/autorest/preprocessor/Preprocessor.java (hunk 2 above)
Comment (alzimmermsft) [reply preprocessor/src/main/java/com/azure/autorest/preprocessor/Preprocessor.java:237]: I want to add it back in the near future to simplify the calls to get List and Map values but testing has shown that using this results in some code generation errors I haven't figured out yet.

**Pull request merged**